extensions/amp-lightbox/1.0/lightbox.js

```diff
@@ -0,0 +1,196 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Preact from '../../../src/preact';
+import {ContainWrapper} from '../../../src/preact/component';
+import {forwardRef} from '../../../src/preact/compat';
+import {
+  useEffect,
+  useImperativeHandle,
+  useLayoutEffect,
+  useRef,
+  useState,
+} from '../../../src/preact';
+import {useStyles} from './lightbox.jss';
+
+const ANIMATION_DURATION = 300;
+const ANIMATION_PRESETS = {
+  'fade-in': [
+    {opacity: 0, visibility: 'visible'},
+    {opacity: 1, visibility: 'visible'},
+  ],
+  'fly-in-top': [
+    {opacity: 0, transform: 'translate(0,-100%)', visibility: 'visible'},
+    {opacity: 1, transform: 'translate(0, 0)', visibility: 'visible'},
+  ],
+  'fly-in-bottom': [
+    {opacity: 0, transform: 'translate(0, 100%)', visibility: 'visible'},
+    {opacity: 1, transform: 'translate(0, 0)', visibility: 'visible'},
+  ],
+};
+
+/**
+ * @param {string} selector
+ * @param {number} time
+ * @return {number}
+ */
+function getElement(selector, time = 5) {
+  return new Promise(async (resolve, reject) => {
+    if (time <= 0) {
+      return reject(`${selector} not found`);
+    }
+    const elements = document.querySelectorAll(selector);
+    if (elements.length) {
+      return resolve(elements);
+    }
+    return setTimeout(async () => await getElement(selector, time - 1), 1000);
+  });
+}
+
+/**
+ * @param {!LightboxDef.Props} props
+ * @param {{current: (!LightboxDef.LightboxApi|null)}} ref
+ * @return {PreactDef.Renderable}
+ */
+function LightboxWithRef(
+  {
+    id,
+    animateIn = 'fade-in',
+    closeButtonAriaLabel,
+    // eslint-disable-next-line no-unused-vars
+    scrollable, // (TODO: discussion)
```

Comment:
It's ok to defer `scrollable` work. Just add it to tracker and remove unused code for now.